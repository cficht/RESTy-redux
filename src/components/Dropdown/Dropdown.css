@import '../../styles/variables.css';

.Dropdown {
  text-align: center;
  padding: 10px;
  width: 100%;
  
  input {
    margin: 10px 10px 0px 10px;
    width: 25%;
  }

  input[name=token] {
    width: 70%;
  }
}