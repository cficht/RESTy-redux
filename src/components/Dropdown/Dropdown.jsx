import React from 'react';
import PropTypes from 'prop-types';

const Dropdown = ({ auth, username, password, token, handleChange }) => {
  const authType = () => {
    if(auth === 'basic')
      return (
        <>
          <input type="text" name="username" value={username} onChange={handleChange} placeholder="username"></input>
          <input type="text" name="password" value={password} onChange={handleChange} placeholder="password"></input>
        </>
      );
    if(auth === 'bearer')
      return (
        <>
          <input type="text" name="token" value={token} onChange={handleChange} placeholder="token"></input>
        </>
      );
  };

  return (
    <>
      <select name="auth" value={auth} onChange={handleChange}>
        <option value="none">None</option>
        <option value="basic">Basic Auth</option>
        <option value="bearer">Bearer Token</option>
      </select>
      <div>
        {authType()}
      </div>
    </>
  );
};

Dropdown.propTypes = {
  auth: PropTypes.string.isRequired,
  username: PropTypes.string.isRequired,
  password: PropTypes.string.isRequired,
  token: PropTypes.string.isRequired,
  handleChange: PropTypes.func.isRequired
};

export default Dropdown;
