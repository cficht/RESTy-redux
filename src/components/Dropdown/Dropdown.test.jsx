import React from 'react';
import { shallow } from 'enzyme';
import Dropdown from './Dropdown';
import { Provider } from 'react-redux';
import store from '../../store';

describe('Dropdown component', () => {
  it('renders Dropdown', () => {
    const wrapper = shallow(
      <Provider store={store}>
        <Dropdown 
          auth="none"
          username="chris"
          password="admin"
          token="232fsdfdsf"
          handleChange={() => {}}/>
      </Provider>
    );
    expect(wrapper).toMatchSnapshot();
  });
});
