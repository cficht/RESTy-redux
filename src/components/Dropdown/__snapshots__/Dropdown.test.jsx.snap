// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dropdown component renders Dropdown 1`] = `
<ContextProvider
  value={
    Object {
      "store": Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      },
      "subscription": Subscription {
        "handleChangeWrapper": [Function],
        "listeners": Object {
          "notify": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        },
        "unsubscribe": null,
      },
    }
  }
>
  <Dropdown
    auth="none"
    handleChange={[Function]}
    password="admin"
    token="232fsdfdsf"
    username="chris"
  />
</ContextProvider>
`;
